src/main/java/dev/openfeature/sdk/OpenFeatureClient.java

```diff
@@ -178,12 +178,6 @@ private <T> FlagEvaluationDetails<T> evaluateFlag(
             // provider must be accessed once to maintain a consistent reference
             provider = stateManager.getProvider();
             ProviderState state = stateManager.getState();
-            if (ProviderState.NOT_READY.equals(state)) {
-                throw new ProviderNotReadyError("provider not yet initialized");
-            }
-            if (ProviderState.FATAL.equals(state)) {
-                throw new FatalError("provider is in an irrecoverable error state");
-            }
```

Comment:
Throwing these before we ran the hooks and setup the `afterHookContext` was the cause of the issue. We just needed to move it later.